File: app/src/main/java/com/mmutert/freshfreezer/data/converters/LocalDateConverter.java
package com.mmutert.freshfreezer.data.converters;

import androidx.room.TypeConverter;

import org.joda.time.LocalDate;
import org.joda.time.LocalDateTime;


public class LocalDateConverter {

    @TypeConverter
    public static LocalDate toDate(String dateLong) {
        return dateLong == null ? null : LocalDate.parse(dateLong);
    }

    @TypeConverter
    public static String fromDate(LocalDate date) {
        return date == null ? null : date.toString();
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/data/converters/LocalDateConverter.kt
package com.mmutert.freshfreezer.data.converters

import androidx.room.TypeConverter
import org.joda.time.LocalDate

object LocalDateConverter {
    @JvmStatic
    @TypeConverter
    fun toDate(dateLong: String?): LocalDate? {
        return if (dateLong == null) null else LocalDate.parse(dateLong)
    }

    @JvmStatic
    @TypeConverter
    fun fromDate(date: LocalDate?): String? {
        return date?.toString()
    }
}